### PR TITLE
Add very basic flutter.yaml documentation.

### PR DESCRIPTION
**Pull request by eseidelGoogle**
Fixes https://github.com/flutter/flutter/issues/1431

@abarth @mpcomplete @qchong

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
I based this off of https://github.com/dart-lang/www.dartlang.org/blob/master/src/site/tools/pub/pubspec.markdown

File: flutter_yaml.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+layout: page
+title: flutter.yaml Format
+permalink: /flutter_yaml
```
Comment (abarth) [inline flutter_yaml.md]: I'd call this `/manifest/`

Also note the trailing `/`

File: flutter_yaml.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+# {{ page.title }}
```
Comment (abarth) [inline flutter_yaml.md]: The template already has this.  You'll end up with two copies if you include it again.

File: flutter_yaml.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+Every Flutter application needs some metadata so it can specify its
+dependencies (services, assets, etc.). Flutter stores this information in a file named `flutter.yaml`, which is written in
```
Comment (abarth) [inline flutter_yaml.md]: Please wrap somewhere near 80 cols

**Comment by abarth** [comment]
LGTM

File: flutter_yaml.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+`name`
+: Required for every application.
+
+`assets`
+: Optional. Relative paths to assets for the application.
+
+`material-design-icons`
+: Optional. Dictionaries specifying icons to include as assets.
+
+`services`
```
Comment (mpcomplete) [inline flutter_yaml.md]: mention the field type? (array of strings)

File: flutter_yaml.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+: Optional. Platform and third-party services (Firebase, Google Cloud Messaging, etc.) to expose to your application.
+
+All other fields are ignored. A simple but complete pubspec looks something
```
Comment (mpcomplete) [inline flutter_yaml.md]: You're missing fields for the update service:

version: x.y.z
update-url: https://example.com/foo/bar/

see https://github.com/flutter/flutter/wiki/On-the-fly-app-updates

**Comment by eseidelGoogle** [comment]
ptal @mpcomplete

File: flutter_yaml.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+{% endprettify %}
+
+## Name
+
+Every application needs a name.  This is used by the flutter tool when
+manipulating files relating to your application on your behalf.
+
+There are no restrictions at this time on valid characters in a name.
+
+## Assets
```
Comment (krisgiesing) [inline flutter_yaml.md]: Link to doc on assets?

**Comment by mpcomplete** [comment]
lgtm

**Comment by eseidelGoogle** [comment]
I just need to check this in and let someone better versed in this take over. :/

**Comment by eseidelGoogle** [comment]
I'm very sorry @qchong and @krisgiesing for not answering your requests.  I'm not a sustainable owner for things like this.

**Pull request merged**